font-types/src/tag.rs

```diff
@@ -6,31 +6,35 @@ use std::{
 
 /// An OpenType tag.
 ///
-/// A tag is a 4-byte array where each byte is in the printable ascii range
-/// (0x20..=0x7E).
+/// [Per the spec][spec], A tag is a 4-byte array where each byte is in the
+/// printable ASCII range (0x20..=0x7E).
+///
+/// We do not strictly enforce this constraint as it is possible to encounter
+/// invalid tags in existing fonts, and these need to be representable.
+///
+/// When creating new tags we encourage ensuring that the tag is valid,
+/// either by using [`Tag::new_checked`] or by calling [`Tag::validate`] on an
+/// existing tag.
+///
+/// [spec]: https://learn.microsoft.com/en-us/typography/opentype/spec/otff#data-types
 #[derive(Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct Tag([u8; 4]);
 
 impl Tag {
-    /// Generate a `Tag` from a byte slice, verifying it conforms to the
-    /// OpenType spec.
-    ///
-    /// The argument must be a non-empty slice, containing at most four
-    /// bytes in the printable ascii range, `0x20..=0x7E`.
+    /// Construct a `Tag` from a byte slice.
     ///
-    /// If the input has fewer than four bytes, it will be padded with spaces
-    /// (`0x20`).
-    ///
-    /// # Panics
-    ///
-    /// This method panics if the tag is not valid per the requirements above.
+    /// If the slice is longer than four bytes, it will be truncated. If it is
+    /// shorter, the missing bytes will be replaced with the space character (0x20).
     pub const fn new(src: &[u8]) -> Tag {
```

Comment:
Can we just change this to take `&[u8; 4]`? That works nicely with typical usage where you pass something like `b"wght"` when you want a const tag. I'm not sure if this breaks any actual uses.

Reply:
Thinking a bit more, there are some places where the short form is used. `CFF` and `cvt` come to mind. I did define the CFF tag as "CFF " when I added it to codegen out of habit. I think skrifa uses `b"cvt"` but that could be changed.

Reply:
honestly I didn't think the `b"xx"` syntax created fixed-size arrays, I thought it always created a static slice? I'm happy to make this change, since the silent truncation behaviour was one of the things I liked the least.

Reply:
Sounds good to me!